sycl/source/detail/jit_compiler.cpp

```diff
@@ -663,6 +663,10 @@ jit_compiler::fuseKernels(QueueImplPtr Queue,
     auto &CG = KernelCmd->getCG();
     assert(CG.getType() == CG::Kernel);
     auto *KernelCG = static_cast<CGExecKernel *>(&CG);
+    if (KernelCG->MKernelIsCooperative) {
```

Comment:
This is a bit too late to check this. Can we have the check performed in the `isFusable` function in `sycl/source/detail/scheduler/commands.cpp` instead?

We could also slightly change the error message shown at the bottom of the changelog of [this file change](https://github.com/intel/llvm/pull/12556/files#diff-f6d023ea4e153531e280d40d990275baa5bf88e699336400232d294b627d0999) and add a test as I'm doing in that PR (under the same file)